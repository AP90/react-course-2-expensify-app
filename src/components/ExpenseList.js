import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";
// funtion required to set up the filter object
import selectExpenses from "../selectors/expenses";


export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No Expenses</p>
            ) : (
                props.expenses.map((item, index) => {
                    return <ExpenseListItem {...item} key={item.id} />
                })
            )
        }
        
    </div>
);


const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
}


export default connect(mapStateToProps)(ExpenseList);
