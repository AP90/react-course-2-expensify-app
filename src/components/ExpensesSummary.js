import React from "react";
import {connect} from "react-redux";
import numeral from "numeral";
import selectExpenses from "../selectors/expenses";
import selectExpensesTotal from "../selectors/expenses-total";

export const ExpensesSummary = ({expenseCount, expensesTotal}) => {
    const expenseWord = expenseCount === 1 ? "expense" : "expenses";
    const formattedExpensesTotal = numeral(expensesTotal / 100).format("$0,0.00");
    return (
        <div>
            <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
        </div>
    )
}




// export class ExpenseSummary extends React.Component {
//     state = {
//         expenseCount: this.props.expenses.length,
//         expensesTotal: expensesTotal(this.props.expenses)
//     }

//     render() {
//         return (
//             <div>
//                 <h3>Viewing {this.state.expenseCount} expenses totalling {numeral(this.state.expensesTotal / 100).format("$0,0.00")}</h3>
//             </div>
//         )
//     }
// };

// const mapStateToProps = (state) => {
//     return {
//         filters: state.filters,
//         expenses: state.expenses
//     }
// };

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    }
};


export default connect(mapStateToProps)(ExpensesSummary);