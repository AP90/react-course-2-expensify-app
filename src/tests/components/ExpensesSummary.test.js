import { ExpensesSummary } from "../../components/ExpensesSummary";
import React from "react";
import {shallow} from "enzyme";
import expenses from "../fixtures/expenses";

test("should show total of one expense", () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235}/>);
    expect(wrapper).toMatchSnapshot();
});

test("should show total of two items", () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={235421110}/>);
    expect(wrapper).toMatchSnapshot();
});