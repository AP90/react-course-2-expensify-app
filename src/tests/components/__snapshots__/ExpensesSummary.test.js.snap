// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show total of one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</div>
`;

exports[`should show total of two items 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $2,354,211.10
  </h1>
</div>
`;
